pkg/apis/auditlog.cattle.io/v1/audit_policy_types.go

```diff
@@ -0,0 +1,131 @@
+
+	// LevelHeaders indicates that along with the default audit log metadata, the request and response headers will
+	// also be included. A LogVerbosity with LevelHeaders is the same as the following LogVerbosity:
+	//
+	// LogVerbosity {
+	//     Request: {
+	//         Headers: true
+	//     },
+	//     Response: {
+	//         Headers: true
+	//     },
+	// }
+	LevelHeaders
+
+	// LevelRequest indicates that along with the default audit log metadata and headers, the request body will also be
+	// included. A LogVerbosity with LevelHeaders is the same as the following LogVerbosity:
+	//
+	// LogVerbosity {
+	//     Request: {
+	//         Headers: true
+	//         Body: true,
+	//     },
+	//     Response: {
+	//         Headers: true
+	//     },
+	// }
+	LevelRequest
+
+	// LevelRequestResponse indicates that along with the default audit log metadata and headers, the request and
+	// response bodies will also be included. A LogVerbosity with LevelHeaders is the same as the following
+	// LogVerbosity:
+	//
+	// LogVerbosity {
+	//     Request: {
+	//         Headers: true
+	//         Body: true,
+	//     },
+	//     Response: {
+	//         Headers: true
+	//         Body: true,
+	//     },
+	// }
+	LevelRequestResponse
+)
+
+// LogVerbosity defines what is included in an audit log. Log metadata (includeing RequestURI, user info, etc) is always present.
+type LogVerbosity struct {
+	// Level is carried over from the previous implementation of audit logging, and provides a shorthand for defining
+	// LogVerbosities. When Level is not LevelNull, Request and Reponse are ignored.
+	Level Level `json:"level"`
+
+	Request  Verbosity `json:"request,omitempty"`
+	Response Verbosity `json:"response,omitempty"`
+}
+
+// +genclient
+// +kubebuilder:printcolumn:name="Enabled",type=string,JSONPath=`.spec.enabled`
+// +kubebuilder:printcolumn:name="Active",type=string,JSONPath=`.status.conditions[?(@.type == "Active")].status`
+// +kubebuilder:printcolumn:name="Valid",type=string,JSONPath=`.status.conditions[?(@.type == "Valid")].status`
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type AuditPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AuditPolicySpec   `json:"spec"`
+	Status AuditPolicyStatus `json:"status,omitempty"`
+}
+
+type AuditPolicySpec struct {
+	Enabled bool `json:"enabled"`
+
+	// Filters described what are explicitly allowed and denied. Leave empty if all logs should be allowed.
+	Filters []Filter `json:"filters,omitempty"`
+
+	// AdditionalRedactions details additional informatino to be redacted. These redactions are only applied to logs
+	// that are allowed by the defiend Filters. Note that if no filters are defined, these redactions will apply to all
+	// logs.
+	AdditionalRedactions []Redaction `json:"additionalRedactions,omitempty"`
+
+	Verbosity LogVerbosity `json:"verbosity,omitempty"`
```

Comment:
There should be some docs for this as well imo. Does the verbosity apply to only the logs "allowed" in this AuditPolicy resource, or does it apply to all of them? If the latter, how are these values merged?

eg: 

- policy 1 has `.verbosity.level = null`
- policy 2 has `.verbosity.level = headers`
- policy 3 has `.verbosity.level = request`